packages/timeline-state-resolver/src/integrations/osc/index.ts

```diff
@@ -340,14 +184,33 @@ export class OSCMessageDevice extends DeviceWithState<OSCDeviceState, DeviceOpti
 			return Promise.resolve()
 		}
 	}
-	private _defaultOscSender(msg: osc.OscMessage, address?: string | undefined, port?: number | undefined): void {
-		this.emitDebug('sending ' + msg.address)
+
+	get connected(): boolean {
+		return this._oscClientStatus === 'connected'
+	}
+	getStatus(): Omit<DeviceStatus, 'active'> {
+		if (this.options?.type === OSCDeviceType.TCP) {
+			return {
+				statusCode: this._oscClientStatus === 'disconnected' ? StatusCode.BAD : StatusCode.GOOD,
+				messages: this._oscClientStatus === 'disconnected' ? ['Disconnected'] : [],
+			}
+		}
+		return {
+			statusCode: StatusCode.GOOD,
+			messages: [],
+		}
+	}
+
+	actions: Record<string, (id: string, payload: Record<string, any>) => Promise<ActionExecutionResult>> = {}
+
+	private _oscSender(msg: osc.OscMessage, address?: string | undefined, port?: number | undefined): void {
+		this.emit('debug', 'sending ' + msg.address)
 		this._oscClient.send(msg, address, port)
 	}
 	private runAnimation() {
 		for (const addr in this.transitions) {
 			// delete old tweens
-			if (this.transitions[addr].started + this.transitions[addr].transition!.duration < this.getCurrentTime()) {
+			if (this.transitions[addr].started + this.transitions[addr].transition!.duration < Date.now()) {
```

Comment:
Do we not want `this.getCurrentTime()` anymore?

Reply:
I'm undecided. I'm not convinced we used this.getCurrentTime in any meaningful way before but I know a lot of the Date.now usage just snuck in while typing away.